docs/keyword-args.rst

```diff
@@ -0,0 +1,149 @@
+
+.. code:: python
+
+    # This is not valid, and will not work.
+    account.create_course(course[name]='Example Course')
+
+What Canvas is effectively doing with the bracket format is grouping things into objects. To achieve a similar effect in Python, CanvasAPI uses dictionaries.
+
+Given an existing :class:`canvasapi.account.Account` object, :code:`account`, the parameter :code:`course[name]` can be passed like this:
+
+.. code:: python
+
+    account.create_course(course={'name': 'Example Course'})
+
+In the background, CanvasAPI will combine the keys of the dictionary with the keyword of the argument, and ultimately send the correct variable to Canvas.
+
+The benefit of this pattern is that multiple parameters with the same prefix can be sent to the same keyword argument. So to pass the :code:`course[name]`, :code:`course[course_code]`, and :code:`course[is_public]` arguments, it would look like this:
+
+.. code:: python
+
+    account.create_course(
+        course={
+            'name': 'Example Course',
+            'course_code': 'TST1234',
+            'is_public': True
+        }
+    )
+
+Nested parameters work easily alongside basic (and list) parameters. For example, :code:`offer` and :code:`enroll_me`:
+
+.. code:: python
+
+    account.create_course(
+        course={
+            'name': 'Example Course',
+            'course_code': 'TST1234',
+            'is_public': True
+        },
+        enroll_me=True,
+        offer=False
+    )
+
+Complex Parameters
+------------------
+
+The three main types of parameters (basic, list, and nested) from above cover most types of parameters Canvas expects. However, there are some types that look deceptively more complex than they actually are. Below are some examples of how to handle these in CanvasAPI.
+
+Deep Nested Parameters
+~~~~~~~~~~~~~~~~~~~~~~
+
+:func:`canvasapi.user.User.edit` has the parameter :code:`user[avatar][url]`, as shown in the `Canvas Docs for Edit a User <https://canvas.instructure.com/doc/api/users.html#method.users.update>`_. Any parameter that takes the form of :code:`foo[bar1][bar2]` with multiple bracketed sub-parameters follows the same rules as normal nested parameters, but with deeper nesting.
+
+.. code:: python
+    
+    user.edit(
+        user={
+            'avatar': {
+                'url': 'http://example.com/john_avatar.png'
+            }
+        }
+    )
+
+List of Nested Parameters
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+:func:`canvasapi.account.Account.add_grading_standards` has the parameters :code:`grading_scheme_entry[][name]` and `grading_scheme_entry[][value]`, as shown in the `Canvas Docs for Create a New Grading Standard <https://canvas.instructure.com/doc/api/grading_standards.html#method.grading_standards_api.create>`_. Any parameter that takes the form of :code:`foo[][bar]` can be represented by a list of dictionaries.
```

Comment:
needs to be code:`grading_scheme_entry[][value]`.

Reply:
Good catch. Fixed.